src/Lerp.cpp

```diff
@@ -143,11 +143,24 @@ Expr lower_lerp(Type final_type, Expr zero_val, Expr one_val, const Expr &weight
                 // ((x + 128) / 256 + x + 128) / 256. Note that
                 // overflow is impossible here because the most our
                 // prod_sum can be is 255^2.
-                if (target.arch == Target::X86) {
-                    // On x86 we have no rounding shifts but we do
-                    // have a multiply-keep-high-half. So it's
-                    // actually one instruction cheaper to do the
-                    // division directly.
+                //
```

Comment:
This isn't going to handle mixed host/device code. I guess the Target argument isn't quite and the interface or the lowering needs a small rethink.

Reply:
> This isn't going to handle mixed host/device code. 

True, but in that case, usually the device code is much more important and the host code can afford slight suboptimal bits.

> I guess the Target argument isn't quite right and the interface or the lowering needs a small rethink.

Yeah, that's definitely true. In hindsight, perhaps having GPU code be features hanging off the host is problematic for cases like this, but that's a nontrivial rewrite, to say the least. 

(Xtensa should almost certainly be its own arch though; I am going to prep a PR to propose that)

Reply:
Actually I think when a hexagon kernel is being compiled, the target arch is Hexagon, due to the way offload works (the submodule gets its own target). We should probably have GPU targets do something similar.

Reply:
(so that target.arch == Target::X86 really really means we're compiling code for x86 right now)